core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java

```diff
@@ -94,6 +102,37 @@ public void createDefaultProcessingStrategyIfNoneSpecified() throws Exception {
                is(instanceOf(new TransactionAwareWorkQueueProcessingStrategyFactory().getProcessingStrategyType())));
   }
 
+  @Test
+  public void processingStrategySetBySystemPropertyOverridesDefault() throws Exception {
+    testWithSystemProperty("org.mule.runtime.core.api.processor.strategy.ProcessingStrategyFactory",
+                           TestProcessingStrategyFactory.class.getName(), () -> {
+                             MuleConfiguration muleConfiguration = new DefaultMuleConfiguration();
+                             when(muleContext.getConfiguration()).thenReturn(muleConfiguration);
+                             createFlow(null);
+                             assertEquals(flow.getProcessingStrategy().getClass(), TestProcessingStrategy.class);
+                           });
+  }
+
+
+  public static class TestProcessingStrategyFactory implements ProcessingStrategyFactory {
```

Comment:
If you extend `AbstractProcessingStrategy` or an existing implementation then this would be bit more concise.